homework2/fastq_filtrator.py

```diff
@@ -0,0 +1,75 @@
+"""GC-check function"""
+def GC_count_fun(f, gc_bounds = (0, 100)):
+  cnt=0
+  #Counting the amount of cytosine and guanine in a string
+  for nucl in f:
+    if (nucl == 'C') or (nucl == 'G'):
+      cnt += 1
+  #Checking and defining boundaries
+  if type(gc_bounds) == tuple:
+    right_margin = gc_bounds[1]
+    left_margin = gc_bounds[0]
+  else:
+    right_margin = gc_bounds
+    left_margin = 0
+
+  if right_margin >= ((cnt/len(f))*100) >= left_margin:
+     return 1
+  else:
+    return 0
+
+"""Read quality check function"""
+def quality_fun(f, quality_threshold = 0):
+  phred_value=0
+  for symbol in f:
+    phred_value += (ord(symbol) - 33)
+  if quality_threshold <= (phred_value/len(f)):
+    return 1
+  else:
+    return 0
+
+"""Length check function"""
+def len_fun(f, length_bounds = (0, 2**32)):
+  #Checking and defining boundaries
+  if type(length_bounds) == tuple:
+    right_margin = length_bounds[1]
+    left_margin = length_bounds[0]
+  else:
+    right_margin = length_bounds
+    left_margin = 0
+
+  if right_margin >= len(f) >= left_margin:
+      return 1
+  else:
+      return 0
+
+"""Main function"""
+def main(input_fastq, output_file_prefix, gc_bounds = (0, 100), length_bounds =  (0, 2**32), quality_threshold = 0, save_filtered = False):
+  #Opening an input file and writing it to the variable f with removing \n at the ends
+  with open(input_fastq, 'r') as input:
+      f = input.read().split('\n')
+      #Checking whether files need to be saved
+      if save_filtered == True:
+        with open((str(output_file_prefix) + "_passed.fastq"), 'w') as output_passed, open((str(output_file_prefix) + "_failed.fastq"), 'w') as output_failed:
```

Comment:
Лишние сравнения и преобразования
```suggestion
      if save_filtered:
        with open(output_file_prefix + "_passed.fastq", 'w') as output_passed, open(output_file_prefix + "_failed.fastq", 'w') as output_failed:
```